sentry-core/src/test/java/io/sentry/core/transport/HttpTransportTest.kt

```diff
@@ -297,6 +297,66 @@ class HttpTransportTest {
         assertTrue(transport.isRetryAfter("event"))
     }
 
+    @Test
+    fun `parse X-Sentry-Rate-Limit and ignore unknown categories`() {
+        val transport = fixture.getSUT()
+
+        whenever(fixture.connection.inputStream).thenThrow(IOException())
+        whenever(fixture.connection.getHeaderField(eq("X-Sentry-Rate-Limits")))
+            .thenReturn("60:default;wtf;error;security:organization")
+
+        val event = SentryEvent()
+
+        transport.send(event)
+        Thread.sleep(2000)
+        assertFalse(transport.isRetryAfter("wtf"))
+    }
+
+    @Test
+    fun `When all categories is set but expired, applies only for specific category`() {
+        val transport = fixture.getSUT()
+
+        whenever(fixture.connection.inputStream).thenThrow(IOException())
+        whenever(fixture.connection.getHeaderField(eq("X-Sentry-Rate-Limits")))
+            .thenReturn("1::key, 60:default;error;security:organization")
+
+        val event = SentryEvent()
+
+        transport.send(event)
+        Thread.sleep(2000)
```

Comment:
`l`: As already pointed out testing with `Thread.sleep` is not the best idea. Imagine we want to test a maximum timeout of one day. How would we do that? Furthermore, this slows down the test suite a lot if we do it in multiple tests. We don't have to improve this in this PR, but maybe we do it at some point.

Reply:
https://github.com/getsentry/sentry-android/pull/381/files#r419898438